compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1528,56 +1541,30 @@ void OMR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
             }
          }
 
-      // -------------------------------------------------------------------
-      // Determine element size and array stride
-      // -------------------------------------------------------------------
-      if (isStringCompressedArrayCopy)
+      if (srcArrayLength)
```

Comment:
To reduce reference arraycopy one of two things has to be true, either `primitiveArray1 || primitiveArray2` or `referenceArray1 || referenceArray2`. The above logic I added covers both cases and properly sets `type` and `elementSize`. Thus `elementSize` will always be non-zero at this point which is why I removed the check.